controller.py

```diff
@@ -9,80 +9,98 @@ def __init__(self,path):
         self.path = path
         self.shortlist = load_shortlist(path)
         self.current_applicant = None
-        self.current_view = "home"
+        self.current_criterion = None
+        self.current_score = None
         self.view = View()
+        self.options = None # intial state
+        self.options_home = {"r":self.show_role_info,
+                             "a":self.show_applicants_list}
+        self.options_applicant_list = {"b":self.show_boot_message,
+                                       "d":self.show_applicant_details}
+        self.options_applicant_detail = {"a":self.show_applicants_list,
+                                         "e":self.edit_score_start,
+                                         "b":self.show_boot_message,
+                                         "O":self.open_applicant_pdf}
 
-    def show_boot_message(self):
+    def show_boot_message(self,k=None):
         """Shortlist overview"""
         self.view.boot_message(self.path,len(self.shortlist.applicants))
-        self.current_view = "home"
+        self.options = self.options_home
 
-    def show_criteria(self):
+    def show_criteria(self,k=None):
         """Display criteria information"""
         self.view.view_criteria(self.shortlist.role,self.shortlist.role.criteria)
-        self.current_view = "criteria"
 
-    def show_role_info(self):
+    def show_role_info(self,k=None):
         """Display role information"""
         self.view.view_role(self.shortlist.role)
-        self.current_view = "role"
 
-    def show_applicants_list(self):
+    def show_applicants_list(self,k=None):
         """List all applicants"""
         self.view.view_applicants_list(self.shortlist)
-        self.current_view = "applicants_list"
+        self.options = self.options_applicant_list
 
-    def show_applicant_details(self):
+    def show_applicant_details(self,k=None):
         """Select an applicant and view details"""
         try:
             i = int(input("Please enter the applicant number:"))
 
             self.current_applicant = self.shortlist.applicants[i-1]
             self.view.view_applicant_details(self.current_applicant)
-            self.current_view = "applicant_details"
+            self.options = self.options_applicant_detail
         except (ValueError, IndexError):
             pass
 
-    def open_applicant_pdf(self):
+    def open_applicant_pdf(self,k=None):
         """Open current applicant's CV"""
         startfile(self.current_applicant.cv)
 
+    def edit_score_start(self,k=None):
+        """select a criteria to edit score for"""
+        self.view.view_criteria(self.shortlist.role,self.shortlist.role.criteria)
+        options = [str(i) for i in range(len(self.shortlist.role.criteria))]
+        self.options = {i:self.edit_criteria_select for i in options} #generate options based on how many criteria there are
```

Comment:
Just as you've used list comprehensions, you can also create a dict using a single statement with "dict comprehensions". It would look like this:

self.options = {c.name: self.edit_criteria_select for c in self.shortlist.role.criteria}

Reply:
I'm slightly confused but I did use dictionary comprehension here unless I'm mistaken?

Reply:
Ah, sorry - you did. My example perhaps was not good. You don't need two separate lines. You can do

Reply:
Or, even,

```
self.options = {str(n): self.edit_criteria_select for n, _ in enumerate(self.shortlist.role.criteria)}
```

`_` is stylistically used in Python as "ignore this result"